checkbox-support/checkbox_support/scripts/run_watcher.py

```diff
@@ -93,124 +90,171 @@ def run(self):
-            for key in self.FLAG_DETECTION["driver"]:
-                if self.FLAG_DETECTION["driver"][key]:
-                    driver = key
-            logger.info("{} was inserted {} controller".format(device, driver))
-            logger.info("usable partition: {}".format(self.MOUNTED_PARTITION))
-            # judge the detection by the expection
+            logger.info(
+                "{} was inserted {} controller".format(
+                    self.device.value, self.driver.value
+                )
+            )
+            logger.info("usable partition: {}".format(self.mounted_partition))
+            # judge the detection by the expectation
             if (
                 self.args.storage_type == "usb2"
-                and device == "new high-speed USB device number"
+                and self.device == USBStorage.Device.HIGH_SPEED_USB
             ):
                 logger.info("USB2 insertion test passed.")
-
-            if self.args.storage_type == "usb3" and (
-                device
-                in (
-                    "new SuperSpeed USB device number",
-                    "new SuperSpeed Gen 1 USB device number",
-                )
-            ):
+            elif self.args.storage_type == "usb3" and self.device in [
+                USBStorage.Device.SUPER_SPEED_USB,
+                USBStorage.Device.SUPER_SPEED_GEN1_USB,
+            ]:
                 logger.info("USB3 insertion test passed.")
+            else:
+                sys.exit("Wrong USB type detected.")
 
             # backup the storage info
-            storage_info_helper(
+            self._storage_info_helper(
                 reserve=True,
                 storage_type=self.args.storage_type,
-                mounted_partition=self.MOUNTED_PARTITION,
+                mounted_partition=self.mounted_partition,
             )
             sys.exit()
 
     def report_removal(self):
-        if self.FLAG_DETECTION["removal"]["USB disconnect, device number"]:
+        if self.action == USBStorage.Action.REMOVAL:
             logger.info("Removal test passed.")
 
-        # remove the storage info
-        storage_info_helper(reserve=False, storage_type=self.args.storage_type)
-        sys.exit()
+            # remove the storage info
+            self._storage_info_helper(
+                reserve=False, storage_type=self.args.storage_type
+            )
+            sys.exit()
 
     def _get_partition_info(self, line_str):
         """get partition info."""
         # looking for string like "sdb: sdb1"
         part_re = re.compile("sd\w+:.*(?P<part_name>sd\w+)")
         match = re.search(part_re, line_str)
         if match:
-            self.MOUNTED_PARTITION = match.group("part_name")
+            self.mounted_partition = match.group("part_name")
 
     def _refresh_detection(self, line_str):
         """
-        refresh values of the dictionary FLAG_DETECTION.
+        refresh values with the lines from journal.
```

Comment:
Please update this docstring, my best understanding is that tries, incrementally, line by line, to see if there is something relevant to this class in the journal. If there is, `self` is updated with the new info